src/main/java/com/ss/training/lms/controller/LibrarianController.java

```diff
@@ -25,139 +18,116 @@
+    //                 })
+    // public ResponseEntity<BookCopies> getAnEntryOfBookCopies(@PathVariable int branch, @PathVariable int book)
+    // {
+    //     HttpStatus status = HttpStatus.OK;
+    //     BookCopies bookCopies = librarianService.getAnEntryOfBookCopies(branch, book);
+    //     if (bookCopies == null)
+    //     {
+    //         status = HttpStatus.NOT_FOUND;
+    //     }
+    //     return new ResponseEntity<BookCopies>(bookCopies , status);
+
+    // }
+
+    // @GetMapping(path="lms/librarian/branches/{branchId}/books")
+    // public ResponseEntity<List<Book>>getBooksAtABranch(@PathVariable int branchId)
+    // {
+    //     List<Book> books = null;
+    //     HttpStatus status = HttpStatus.OK;
+
+    //     try {
+    //         books = librarianService.getBooksAtABranch(branchId);
+    //         if (books == null)
+    //         {
+    //             status = HttpStatus.NOT_FOUND;
+    //         }
+    //     } catch (SQLException e) {
+    //         e.printStackTrace();
+    //         return new ResponseEntity<List<Book>>(books, HttpStatus.INTERNAL_SERVER_ERROR);
+
+    //     }
+    //     return new ResponseEntity<List<Book>>(books, status);
+
+    // }
 
     @PutMapping(value = "/lms/librarian/branches/{branchId}/books/{bookId}")
-    public ResponseEntity<BookCopies> updateCopies(@PathVariable int branchId,
+    public ResponseEntity<String> updateCopies(@PathVariable int branchId,
                                                @PathVariable int bookId,
                                                @RequestBody BookCopies bookCopies)
         {
         HttpStatus status = HttpStatus.OK;
         if (bookCopies == null)
         {
             status = HttpStatus.BAD_REQUEST;
-            return new ResponseEntity<BookCopies>(bookCopies, status);
-        }
-
-        try {
-            boolean updated = librarianService.updateCopies(bookCopies);
-            if (updated == false)
-            {
-            	status = HttpStatus.BAD_REQUEST;
-                return new ResponseEntity<BookCopies>(bookCopies, status);
-            }
-        } catch (SQLException e) {
-            e.printStackTrace();
-            status = HttpStatus.INTERNAL_SERVER_ERROR;
-            return new ResponseEntity<BookCopies>(bookCopies, status);
+            return new ResponseEntity<String>("Problem with request", status);
         }
-        return new ResponseEntity<BookCopies>(bookCopies, status);
+        librarianService.updateCopies(bookCopies);
+        return new ResponseEntity<String>("success", status);
     }
 
-    @PutMapping(value = "/lms/librarian/branches/{branchId}")
+    @PutMapping(value = "/lms/librarian/branches/{branchId}/update")
     @ResponseStatus(HttpStatus.OK)
-    public ResponseEntity<LibraryBranch> updateBranch(@PathVariable int branchId,
+    public ResponseEntity<String> updateBranch(@PathVariable int branchId,
                              @RequestBody LibraryBranch libraryBranch){
```

Comment:
Similarly, the branchId in the path for this method isn't used either. The branch that gets updated is always the one with the ID of the one that gets passed in the request body.